PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -499,6 +522,107 @@ void BookWeightsHistograms()
+  Double_t vvvariableMinMax[5][2] = { {0.,1.}, // integrated
+                                      {fMultiplicityBins[1],fMultiplicityBins[2]}, // multiplicity
+                                      {fCentralityBins[1],fCentralityBins[2]}, // centrality
+                                      {fKinematicsBins[PT][1],fKinematicsBins[PT][2]},
+                                      {fKinematicsBins[ETA][1],fKinematicsBins[ETA][2]}
+                                    };
+  */
+
+  TString vvVariable[3] = {"integrated", "multiplicity", "centrality"};
+
+  for (Int_t k = 0; k < 4; k++) // order [2p=0,4p=1,6p=2,8p=3]
+  {
+    for (Int_t n = 0; n < 6; n++) // harmonic [n=1,n=2,...,n=6]
+    {
+      for (Int_t v = 0; v < 3; v++) // variable [0=integrated,1=vs. multiplicity,2=vs. centrality]
+      {
+
+        // if(PTKINE == v  && !fCalculatePtCorrelations){continue;}
+        // if(ETAKINE == v  && !fCalculateEtaCorrelations){continue;}
+
+        /*
+        // per demand, custom meeting for kine dependence:
+        if(PTKINE == v  && fUseCustomKineDependenceBins[PTq])
+        {
+         fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]",k,n,v),Form("#LT#LTcos[%s(%s)]#GT#GT",1==n+1?"":Form("%d",n+1),oVariable[k].Data()),fKineDependenceBins[PTq]->GetSize()-1,fKineDependenceBins[PTq]->GetArray());
+        }
+        else if(ETAKINE == v  && fUseCustomKineDependenceBins[ETAq])
+        {
+         fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]",k,n,v),Form("#LT#LTcos[%s(%s)]#GT#GT",1==n+1?"":Form("%d",n+1),oVariable[k].Data()),fKineDependenceBins[ETAq]->GetSize()-1,fKineDependenceBins[ETAq]->GetArray());
+        }
+        else
+        {
+        */
+        // the default binning:
+        // fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]",k,n,v),Form("#LT#LTcos[%s(%s)]#GT#GT",1==n+1?"":Form("%d",n+1),oVariable[k].Data()),vvvariableNBins[v],vvvariableMinMax[v][0],vvvariableMinMax[v][1]);
+        nl_a.fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]", k, n, v), Form("#LT#LTcos[%s(%s)]#GT#GT", 1 == n + 1 ? "" : Form("%d", n + 1), oVariable[k].Data()), 2000, 0., 2000.);
+        // } // else
+        nl_a.fNestedLoopsPro[k][n][v]->SetStats(kFALSE);
+        nl_a.fNestedLoopsPro[k][n][v]->Sumw2();
+        nl_a.fNestedLoopsPro[k][n][v]->GetXaxis()->SetTitle(vvVariable[v].Data());
+        // fNestedLoopsPro[k][n][v]->SetFillColor(colorsW[v]-10);
+        // fNestedLoopsPro[k][n][v]->SetLineColor(colorsW[v]);
+        /*
+        if(fUseFixedNumberOfRandomlySelectedParticles && 1==v) // just a warning for the meaning of multiplicity in this special case
+        {
+         nl_a.fNestedLoopsPro[k][n][1]->GetXaxis()->SetTitle("WARNING: for each multiplicity, fFixedNumberOfRandomlySelectedParticles is selected randomly in Q-vector");
+        }
+        */
+
```

Comment:
Please, don't leave code commented
In these initial phases everyone is looking at what others are doing 
Leaving code commented makes difficult to understand what the code is doing